dev/AppLifecycle/AppLifecycle.idl

```diff
@@ -1,11 +1,79 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-namespace Microsoft.ApplicationModel.Activation
+namespace Microsoft.Windows.AppLifecycle
 {
-    static runtimeclass AppLifecycle
+    enum ExtendedActivationKind
     {
-        static Windows.ApplicationModel.Activation.IActivatedEventArgs GetActivatedEventArgs();
+        Launch = 0,
+        Search,
+        ShareTarget,
+        File,
+        Protocol,
+        FileOpenPicker,
+        FileSavePicker,
+        CachedFileUpdater,
+        ContactPicker,
+        Device,
+        PrintTaskSettings,
+        CameraSettings,
+        RestrictedLaunch,
+        AppointmentsProvider,
+        Contact,
+        LockScreenCall,
+        VoiceCommand,
+        LockScreen,
+        PickerReturned = 1000,
+        WalletAction,
+        PickFileContinuation,
+        PickSaveFileContinuation,
+        PickFolderContinuation,
+        WebAuthenticationBrokerContinuation,
+        WebAccountProvider,
+        ComponentUI,
+        ProtocolForResults = 1009,
+        ToastNotification,
+        Print3DWorkflow,
+        DialReceiver,
+        DevicePairing,
+        UserDataAccountsProvider,
+        FilePickerExperience,
+        LockScreenComponent,
+        ContactPanel,
+        PrintWorkflowForegroundTask,
+        GameUIProvider,
+        StartupTask,
+        CommandLineLaunch,
+        BarcodeScannerProvider,
+        PrintSupportJobUI,
+        PrintSupportSettingsUI,
+        PhoneCallActivation,
+        VpnForeground,
+
+        // NOTE: Values below 5000 are designated for the platform.  The above list is kept in sync with
```

Comment:
> // NOTE: Values below 5000 are designated for the platform.  The above list is kept in sync with [](http://example.com/codeflow?start=7&length=97)

Does the OS promise to only use values <5000?

(Can we look forward to such a comment in an OS header sometime soon?)